extensions/blocks/repeat-visitor/components/edit.js

```diff
@@ -77,9 +78,7 @@ class RepeatVisitorEdit extends Component {
 					<TextControl
 						className="wp-block-jetpack-repeat-visitor-threshold"
 						defaultValue={ this.props.attributes.threshold }
```

Comment:
Oh, took me a while to realize we're using `defaultValue` (rather than `value`) here. That's a bit unusual :thinking: What about changing this to a regular `value`? Is it possible that we could drop `state` handling altogether then? (We will probably lose the `Please enter a valid number` message in the process, but users simply wouldn't be able to enter any forbidden chars, which seems good enough -- and pretty standard behavior.)

Reply:
Yes, we could use `value`, drop the `help` property, and completely remove the internal state of the component. However, I think this reduction in code complexity comes at a high UX cost where we silently block user interaction without visible feedback.

The initial implementation for this actually used `value` and we`d received internal feedback during our CFT; I can't imagine how much worse it'd be for the general population.